### PR TITLE
Integrate LDAP Proxy

### LINKED ISSUE #28: Integrate LDAP proxy
Make it possible to configure the [LDAP proxy](https://github.com/NetKnights-GmbH/privacyidea-ldap-proxy).

### PR DESCRIPTION
**Pull request by fredreichbier**
Should fix #28 for now.

Hmm, sadly this is a lot of code and also includes a lot of refactoring ... :-/

### CODE REVIEW

File: authappliance/lib/extdialog.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import dialog
```
Comment (cornelinux) [inline authappliance/lib/extdialog.py]: Can you please add a file header

File: authappliance/lib/ldap_proxy.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
+import socket
```
Comment (cornelinux) [inline authappliance/lib/ldap_proxy.py]: Please add a file header according to other ``lib`` files.

File: authappliance/pi-appliance [hunk 3]
```diff
@@ -1132,109 +1456,90 @@ class MainMenu(object):
...
-                elif tags.startswith("Webserver"):
-                    self.webserverDialog.menu()
-                elif tags.startswith("Database"):
-                    self.dbDialog.menu()
-                elif tags.startswith("Audit"):
-                    self.auditDialog.menu()
-                    
+                                     item_help=True)
+            if menu is not None:
+                menu()
```
Comment (cornelinux) [inline authappliance/pi-appliance]: I like this change! :-)

File: authappliance/lib/extdialog.py (hunk 1 above)
Comment (fredreichbier) [reply authappliance/lib/extdialog.py]: done!

File: authappliance/lib/ldap_proxy.py (hunk 2 above)
Comment (fredreichbier) [reply authappliance/lib/ldap_proxy.py]: done!

**Comment by cornelinux** [comment]
I merge u merge?

**Pull request merged**